lib/views/insert-image-view.coffee

```diff
@@ -169,14 +169,20 @@ class InsertImageView extends View
 
   copyImage: (file, callback) ->
     return callback() if utils.isUrl(file) || !fs.existsSync(file)
+    destFileName = path.basename(file)
 
     try
-      destFile = path.join(@siteLocalDir(), @siteImagesDir(), path.basename(file))
+      if config.get("postAssetFolder")
+        destFile = path.join(path.dirname(@editor.getPath()),
+                             templateHelper.parseFileSlug(@editor.getPath()),
+                             destFileName)
```

Comment:
Have you tried to change config siteImagesDir to `"_posts/{year}/{slug}/"`? Probably no need to have this if case.